docs/batch-system-integration.md

```diff
@@ -0,0 +1,839 @@
+Writing Routes For HTCondor-CE
+==============================
+
+The [JobRouter](http://research.cs.wisc.edu/htcondor/manual/v8.6/5_4HTCondor_Job.html) is at the heart of HTCondor-CE and allows admins to transform and direct jobs to specific batch systems. Customizations are made in the form of job routes where each route corresponds to a separate job transformation: If an incoming job matches a job route's requirements, the route creates a transformed job (referred to as the 'routed job') that is then submitted to the batch system. The CE package comes with default routes located in `/etc/condor-ce/config.d/02-ce-*.conf` that provide enough basic functionality for a small site.
+
+If you have needs beyond delegating all incoming jobs to your batch system as they are, this document provides examples of common job routes and job route problems.
+
+!!! note "Definitions"
+    - **Incoming Job**: A job which was submitted to the CE from an outside source, such as a GlideinWMS Factory.
+
+    - **Routed Job**: A job which was transformed by the JobRouter.
+
+    - **Batch System**: The underlying batch system that the HTCondor-CE will submit.  This can be Slurm, PBS, HTCondor, SGE, LSF,...
+
+Quirks and Pitfalls
+-------------------
+
+-   If a value is set in [JOB\_ROUTER\_DEFAULTS](#job_router_defaults) with `eval_set_<variable>`, override it by using `eval_set_<variable>` in the `JOB_ROUTER_ENTRIES`. Do this at your own risk as it may cause the CE to break.
+-   Make sure to run `condor_ce_reconfig` after changing your routes, otherwise they will not take effect.
+-   HTCondor batch system only: Local universe jobs are excluded from any routing.
+
+How Job Routes are Constructed
+------------------------------
+
+Each job route’s [ClassAd](http://research.cs.wisc.edu/htcondor/manual/v8.6/4_1HTCondor_s_ClassAd.html) is constructed by combining each entry from the `JOB_ROUTER_ENTRIES` with the `JOB_ROUTER_DEFAULTS`. Attributes that are [set_*](#setting-attributes) in `JOB_ROUTER_ENTRIES` will override those [set_*](#setting-attributes) in `JOB_ROUTER_DEFAULTS`
+
+### JOB_ROUTER_ENTRIES
```

Comment:
I forget if you need to backslash the underline in headers...

Reply:
Just checked, doesn't look necessary